src/util/mod.rs

```diff
@@ -347,16 +430,75 @@ pub fn file_icon_suffix<P: AsRef<Path>>(f: P, is_file: bool) -> &'static str {
 ///
 /// The specified path must point to a file.
 pub fn get_raw_fs_metadata<P: AsRef<Path>>(f: P) -> RawFileData {
-    let f = f.as_ref();
+    get_raw_fs_metadata_impl(f.as_ref())
+}
+
+fn get_raw_fs_metadata_impl(f: &Path) -> RawFileData {
+    let meta = f.metadata().expect("Failed to get requested file metadata");
     RawFileData {
         mime_type: guess_mime_type_opt(f).unwrap_or_else(|| if file_binary(f) {
             "application/octet-stream".parse().unwrap()
         } else {
             "text/plain".parse().unwrap()
         }),
         name: f.file_name().unwrap().to_str().expect("Failed to get requested file name").to_string(),
-        last_modified: file_time_modified(f),
-        size: f.metadata().expect("Failed to get requested file metadata").len(),
+        last_modified: file_time_modified(&meta),
+        size: file_length(&meta, &f),
         is_file: true,
     }
 }
+
+/// Recursively copy a directory
+///
+/// Stolen from https://github.com/mdunsmuir/copy_dir/blob/0.1.2/src/lib.rs
```

Comment:
Considering that crate looks unstable, just copying the code in makes sense I guess. Might be nice to look to help work on making the crate stable (enough) and eventually swap to using the crate.

Reply:
Well, it's less that i's unstable and more that both it 
![2020-01-24-17:33:25-nabtop-570x717](https://user-images.githubusercontent.com/6709544/73085828-a96f0880-3ecf-11ea-8b27-6c5b6cd65476.png)
and the author/maintainer
![2020-01-24-17:31:48-nabtop-932x292](https://user-images.githubusercontent.com/6709544/73085751-7c225a80-3ecf-11ea-8084-79ff8b3f2db9.png)
seem quite positively dead, so there's approximately no chance of upstreaming anything

Reply:
Ah, fair enough